appshell/appshell_extensions_mac.mm

```diff
@@ -1175,6 +1175,152 @@ int32 SetMenuTitle(CefRefPtr<CefBrowser> browser, ExtensionString command, Exten
     return NO_ERROR;
 }
 
+OSStatus _RunToolWithAdminPrivileges( AuthorizationRef authorizationRef, const std::string& tool, const std::vector<std::string> &args)
+{
+    OSStatus status = 0;
+    if(!authorizationRef || tool.length() ==  0 || args.size() == 0)
+        return errAuthorizationInvalidSet;
+    
+    FILE *pipe = NULL;
+
+    // convert to std::vector<char *> array.
+    std::vector<char*> argv(1 + args.size(), NULL);
+
+    for (size_t i = 0; i < args.size(); ++i)
+        argv[i] = const_cast<char*>(args[i].c_str());
+    
+    // This is a deprecated API. Apple recommends a dedicated helper to
+    // fix this issue. This ideally should be replaced by SMBless API.
+    status = AuthorizationExecuteWithPrivileges(authorizationRef, tool.c_str(),
```

Comment:
[This API was deprecated](https://developer.apple.com/library/mac/documentation/Security/Reference/authorization_ref/#//apple_ref/c/func/AuthorizationExecuteWithPrivileges) all the way back in OS X 10.7... which means it could stop working pretty soon. Would it be safer to use an officially supported technique?  [This Stackoverflow page](http://stackoverflow.com/questions/6841937/authorizationexecutewithprivileges-is-deprecated) lists several alternatives.